pkg/nsx/services/vpc/vpc.go

```diff
@@ -627,7 +631,31 @@ func (s *VPCService) GetVpcConnectivityProfile(nc *common.VPCNetworkConfigInfo,
 	return &vpcConnectivityProfile, nil
 }
 
-func (s *VPCService) CreateOrUpdateVPC(obj *v1alpha1.NetworkInfo, nc *common.VPCNetworkConfigInfo) (*model.Vpc, error) {
+/*
+IsLBProviderChanged is used to judge if the lb provider is changed from day0 to day2
+
+	The lb provider is allowed to be NoneLB in day0, and changed to AIVLB or NSXLB in day2
+	return true if the lb provider is changed
+*/
+func (s *VPCService) IsLBProviderChanged(existingVPC *model.Vpc, lbProvider LBProvider) bool {
+	if existingVPC == nil {
```

Comment:
Is it easier to call this function with the condition that existing VPC is not nil? Just feel it is not necessary to add such check if we can have some pre-check or guidance in the caller.